tests/rptest/services/kafka_cli_producer.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2021 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import re
+import threading
+from ducktape.services.background_thread import BackgroundThreadService
+
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from ducktape.utils.util import wait_until
+
+
+class KafkaCliProducer(BackgroundThreadService):
```

Comment:
Please write an app using kafka driver which writes small messages in a closed loop with linger.ms set to 0. It may print the number of produced records to use the same idea of reading the output to get an idea of the amount of the produced records.

With this test we want
 - to stress the system with a barrage or small batches to increase the chance of having the system doing something during partition movement
 - the test to be so obvious that we can be sure that it does what we want it to do.

The current approach doesn't fit the criteria: it relies on multiple pieces which we don't own glued together. Not only we need to understand the failure modes of the third party software (kafka-console-producer.sh) but we also need to be sure that we glued it correctly with the other parts (dd, in-lined bash), that we handle the errors and that the generated workload overall resembles a typical use-case.

Let's analyze what may go wrong.

`KafkaCliProducer` executes something like `for (( i=0; i < 3; i++ )) ; do dd if=/dev/urandom bs=16b count=1 | ./kafka-console-producer.sh; sleep 1 ; done`

We may expect that once `kafka-console-producer.sh` crashes `KafkaCliProducer` should crash too.

Let's take an a priori crashing app:

```
# python3 -c 'raise Exception("oops")'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
Exception: oops

# echo $?
1
```
And substitute `kafka-console-producer.sh` for it

```
# for (( i=0; i < 3; i++ )) ; do dd if=/dev/urandom bs=16b count=1 | python3 -c "raise Exception(\"$i\")" ; sleep 1 ; done

1+0 records in
1+0 records out
8192 bytes (8.2 kB, 8.0 KiB) copied, 4.8084e-05 s, 170 MB/s
Traceback (most recent call last):
  File "<string>", line 1, in <module>
Exception: 0
1+0 records in
1+0 records out
8192 bytes (8.2 kB, 8.0 KiB) copied, 4.7272e-05 s, 173 MB/s
Traceback (most recent call last):
  File "<string>", line 1, in <module>
Exception: 1
1+0 records in
1+0 records out
8192 bytes (8.2 kB, 8.0 KiB) copied, 5.0668e-05 s, 162 MB/s
Traceback (most recent call last):
  File "<string>", line 1, in <module>
Exception: 2
# echo $?
0
```

It doesn't crash! And it even prints `8192 bytes (8.2 kB, 8.0 KiB) copied, 5.0668e-05 s, 162 MB/s` which we may parse and consider an indicator of written messages to Redpanda.

Another problem with this code is that it works in bursts: it writes 10MB as a single message, sleeps for one second and then repeat the loop. Every moment when the script sleeps it reduces parallelism of the workload and the partition movement and thous reduces the chances of finding the problem.

Another problem is using 10MB as a single message. Using multiple small messages creates more friction in the system and more likely stress it to cause the problem.

Resume: write a simple app using kafka driver to continuously write small messages to redpanda.